packages/react-instantsearch-nextjs/src/InstantSearchNext.tsx

```diff
@@ -47,6 +50,17 @@ export function InstantSearchNext<
   ...instantSearchProps
 }: InstantSearchNextProps<TUiState, TRouteState>) {
   const isMounting = useRef(true);
+  const isServer = typeof window === 'undefined';
+  const pathname = usePathname();
+  const hasRouteChanged =
+    !isServer &&
+    window[InstantSearchLastPath] &&
+    window[InstantSearchLastPath] !== pathname;
+
+  // We only want to trigger a search from a server environment
+  // or if a Next.js route change has happened on the client
+  const shouldTriggerSearch = isServer || hasRouteChanged;
```

Comment:
`InitializePromise` and `TriggerSearch` wouldn't do anything on the client.

You don't need to have them to trigger a search on the client, normal CSR with 2 renders kicks in.

Reply:
That means just having `isServer` is enough I think

Reply:
This breaks in a dynamic route setting, where a route change caused by Next.js will unmount/remount InstantSearchNext, and not trigger the search.

Adding the check for path change allows `<TriggerSearch>` to be mounted and executed as you can see in this recording (the generated sandbox is failing to setup sadly).

I don't exactly understand how it kicks in. Probably Next.js route change performs similarly to `router.refresh()` and executes server components?

https://github.com/user-attachments/assets/8ff0e7cf-5528-4190-b979-05c759dc957f

Reply:
Ah my bad sorry !
Actually I think this solution is nice, we could even choose not to delete `initialResults`and it would avoid that flash of no results. Not sure what the condition would look like though, but it can be done later in another PR.